src/content/docs/dropins/checkout/functions.mdx

```diff
@@ -0,0 +1,714 @@
+```
+
+## redirect
+
+The `redirect` function redirects the user to a specific URL. It takes a single parameter (`url`) of type `string`, which represents the URL to redirect to.
+
+```ts
+export const redirect: (url: string) => void
+```
+
+### Usage
+
+See the following example for usage details:
+
+```ts
+import { redirect } from '@/checkout/api/redirect';
+
+redirect('https://example.com');
+```
+
+## resetCheckout
+
+The `resetCheckout` function resets the checkout process to its initial state by clearing the cart data.
+
+```ts
+export const resetCheckout: () => void
+```
+
+### Usage
+
+See the following example for usage details:
+
+```ts
+import { state } from '@/checkout/lib/state';
+import { cartSignal } from '@/checkout/signals';
+
+resetCheckout();
+```
+
+## resetCustomer
+
+The `resetCustomer` function clears the current customer information and resets the customer state to its inital value in the checkout process.
+
+```ts
+export const resetCustomer: () => void
+```
+
+### Usage
+
+See the following example for usage details:
+
+```ts
+import { state } from '@/checkout/lib/state';
+import { customerSignal } from '@/checkout/signals';
+
+resetCustomer();
+```
+
+## setBillingAddress
+
+The `setBillingAddress` function sets the billing address for a specific cart using the [`setBillingAddressOnCart`](https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/set-billing-address/) Adobe Commerce GraphQL mutation.
+
+```ts
+setBillingAddress({input: BillingAddressInput})
+```
+
+It takes an `input` parameter where `BillingAddressInput` is an interface that contains the specific properties for the billing address:
+
+```ts
+interface BillingAddressInput {
+  address?: AddressInput;
+  customer_address_id?: number;
+  same_as_shipping?: boolean;
+  use_for_shipping?: boolean;
+}
+```
+
+### Returns
+
+The function returns a promise that resolves to the transformed cart data fetched from the API.
+
+```ts
+transformCart
+```
+
+### Usage
+
+To set the billing address as the same as the shipping address:
+
+```ts
+import { setBillingAddress } from '@/checkout/api/setBillingAddress';
+
+setBillingAddress({
+  input: {
+    same_as_shipping: true,
+  }
+});
+```
+
+To set a specific billing address:
+
+```ts
+import { setBillingAddress } from '@/checkout/api/setBillingAddress';
+
+setBillingAddress({
+  input: {
+    address: {
+      firstname: 'John',
+      lastname: 'Doe',
+      street: ['123 Main St', 'Apt 1'],
+      city: 'New York',
+      postcode: '10001',
+      country_code: 'US',
+      telephone: '555-555-5555',
+    },
+    same_as_shipping: false,
+  }
+});
+```
+
+## setGuestEmailOnCart
+
+The `setGuestEmailOnCart` function sets the email address for a guest user on the cart using the [`setGuestEmailOnCart`](https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/set-guest-email/) mutation.
+
+It takes the `email` parameter, which is the email address of the guest.
```

Comment:
Should you have an OptionsTable here?

Reply:
Yes, good catch. It appears that it accepts a single parameter (`email`).